src/test/java/org/wildfly/security/sasl/test/StringPrepTest.java

```diff
@@ -837,16 +837,16 @@ public void testForbidUnassigned() throws Exception {
     // ---------------------- helpers ----------------------
 
     private String codePointToString(int codePoint) {
-        ByteStringBuilder b = new ByteStringBuilder();
-        b.appendUtf8Raw(codePoint);
-        return new String(b.toArray(), StandardCharsets.UTF_8);
+        StringBuffer sb = new StringBuffer();
```

Comment:
Really the best approach is to use Character.toString(codePoint) if it is a non-surrogate pair.  Otherwise you could fall back to a char[2].